packages/next/src/build/webpack/plugins/next-types-plugin/index.ts

```diff
@@ -107,6 +107,19 @@ if ('${method}' in entry) {
       '${method}'
     >
   >()
+  checkFields<
+    Diff<
+      {
+        __tag__: '${method}',
+        __return_type__: Response | Promise<Response>
```

Comment:
@joulev I don't think `ImageResponse` itself is an instance of `Response` so this will break if your `route.ts` is returning an `ImageResponse` (for OG).

This line needs to add: `ImageResponse | Promise<ImageResponse>`
and needs to import it at the top:
`import type { NextRequest, ImageResponse } from 'next/server.js'`

Reply:
cc: @ijjk